core/src/main/java/org/apache/iceberg/avro/BuildAvroProjection.java

```diff
@@ -219,6 +222,7 @@ public Schema map(Schema map, Supplier<Schema> value) {
 
       // element was changed, create a new map
       if (valueSchema != map.getValueType()) {
+        // TODO: we do not copy field ids here
```

Comment:
This is okay because this class assumes that the incoming schema has IDs. If you'd like, we can add Preconditions to check that assumption.

Reply:
That's ok. It was just messing with my testing, making it slightly more complex. I've let comments in my testing code to highlight the problem